### PR TITLE
[Close loop BZ: 1998680]: Object bucket with large number of objects

### PR DESCRIPTION
**Pull request by parikshithb**
Signed-off-by: Parikshith Byregowda <pbyregow@redhat.com>

### CODE REVIEW

File: tests/e2e/scale/noobaa/test_hsbench.py [hunk 1]
```diff
@@ -57,3 +75,28 @@ def test_s3_benchmark_hsbench(self, hsbenchs3):
...
+        """
+        Tests large number of objects on a single bucket:
+        * Create an OBC
+        * Run hs S3 benchmark to create 700k objects on the object bucket
+        * Post writing objects verify OBC creation
+        """
+        # Create an Object bucket
+        object_bucket = bucket_factory(amount=1, interface="OC")[0]
+
+        # Write 1M objects to the object bucket
```
Comment (tiffanyn108) [inline tests/e2e/scale/noobaa/test_hsbench.py:92]: This test is to write 70000 objects, not 1M.

File: tests/e2e/scale/noobaa/test_hsbench.py [hunk 2]
```diff
@@ -3,7 +3,11 @@
 from ocs_ci.ocs import hsbench
 from ocs_ci.utility import utils
 from ocs_ci.framework.testlib import E2ETest, scale
-from ocs_ci.framework.pytest_customization.marks import vsphere_platform_required
+from ocs_ci.framework.pytest_customization.marks import (
+    vsphere_platform_required,
```
Comment (tiffanyn108) [inline tests/e2e/scale/noobaa/test_hsbench.py:7]: This test also can be run on both aws and VMware clusters.

File: tests/e2e/scale/noobaa/test_hsbench.py [hunk 3]
```diff
@@ -57,3 +75,28 @@ def test_s3_benchmark_hsbench(self, hsbenchs3):
...
+        object_bucket = bucket_factory(amount=1, interface="OC")[0]
+
+        # Write 1M objects to the object bucket
+        s3bench.run_benchmark(
+            num_obj=700000,
+            timeout=12000,
+            access_key=mcg_obj.access_key_id,
+            secret_key=mcg_obj.access_key,
+            end_point=f"http://s3.openshift-storage.svc/{object_bucket.name}",
+        )
```
Comment (Akarsha-rai) [inline tests/e2e/scale/noobaa/test_hsbench.py:99]: is it possible to write such that few are small and few are larger objects in 70000/1M objects?

File: tests/e2e/scale/noobaa/test_hsbench.py (hunk 1 above)
Comment (parikshithb) [reply tests/e2e/scale/noobaa/test_hsbench.py]: ack

File: tests/e2e/scale/noobaa/test_hsbench.py (hunk 3 above)
Comment (parikshithb) [reply tests/e2e/scale/noobaa/test_hsbench.py:99]: hs_benchmark is set with default values when initialized, if we want, need to modify the class.
Also for this particular test we don't need multiple ob sizes, we are only focused on object count.

Comment (Akarsha-rai) [reply tests/e2e/scale/noobaa/test_hsbench.py:99]: ack, thanks

File: tests/e2e/scale/noobaa/test_hsbench.py [hunk 4]
```diff
@@ -57,3 +75,28 @@ def test_s3_benchmark_hsbench(self, hsbenchs3):
...
+
+        # Create new OBC and verify it is bound
+        bucket_factory(interface="OC")
```
Comment (hnallurv) [inline tests/e2e/scale/noobaa/test_hsbench.py:102]: How new OBCs get created here?

The https://bugzilla.redhat.com/show_bug.cgi?id=1998680#c38 mentions you created 100 OBCs.

File: tests/e2e/scale/noobaa/test_hsbench.py (hunk 2 above)
Comment (parikshithb) [reply tests/e2e/scale/noobaa/test_hsbench.py:7]: True, I only added skip for existing RGW test for platforms other than vsphere. Skip was at class level.

File: tests/e2e/scale/noobaa/test_hsbench.py (hunk 4 above)
Comment (parikshithb) [reply tests/e2e/scale/noobaa/test_hsbench.py:102]: Does not matter, OBC should be bound instead of going to LOST or PENDING state.

**Review by hnallurv — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Review by tiffanyn108 — APPROVED** [review]

**Pull request merged**